cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -317,8 +318,27 @@ func IsCoreDNSConfigMapMigrationRequired(corefile string) bool {
 	return true
 }
 
-func migrateCoreDNSConfigMap(client clientset.Interface, cm *v1.ConfigMap, corefile, currentInstalledCoreDNSVersion string) error {
-	// Since the current Configuration present is the not the default version, try and migrate the Config.
+func migrateCoreDNSCorefile(client clientset.Interface, cm *v1.ConfigMap, corefile, currentInstalledCoreDNSVersion string) error {
+	// Take a copy of the Corefile data as `Corefile-backup` and update the ConfigMap
+	// Also point the CoreDNS deployment to the `Corefile-backup` data.
+
+	if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Update(&v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kubeadmconstants.CoreDNSConfigMap,
+			Namespace: metav1.NamespaceSystem,
+		},
+		Data: map[string]string{
+			"Corefile":        corefile,
+			"Corefile-backup": corefile,
+		},
+	}); err != nil {
+		return errors.Wrap(err, "unable to update the CoreDNS ConfigMap with backup Corefile")
+	}
+	if err := patchCoreDNSDeployment(client, "Corefile-backup"); err != nil {
```

Comment:
@neolit123 I moved the `Corefile` backup into the same ConfigMap and then patching the deployment to point to `Corefile-backup` here. 
When the CoreDNS deployment is updated to the latest during the upgrade, that step would point the key back to `Corefile`, which will again trigger a rollout. 
Let me know if this logic would work or I should move the "deployment-patch" step into a separate step.

Reply:
thanks for the update, SGTM!